volatility3/framework/layers/qemu.py

```diff
@@ -58,12 +58,15 @@ def _read_configuration(self, base_layer: interfaces.layers.DataLayerInterface,
         data = b''
         for i in range(base_layer.maximum_address, base_layer.minimum_address, -chunk_size):
             if i != base_layer.maximum_address:
-                data = base_layer.read(i, chunk_size) + data
+                data = base_layer.read(i, chunk_size).rstrip(b'\x00') + data
```

Comment:
Whilst I like taking the trailing null bytes into account, this could have unintended consequences (given the stripping may happen simply on a block that we tack more data to.  Since it's liable to be JSON it's not likely to ever exhibit itself, but the unintended consequences just make me slight nervous.  Better to do the `rstrip` at the very end, so it's definitely taking off the end of the JSON not somewhere in the middle.

Reply:
You're right! 👍 I moved it to the end of the line.